bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/ProcessJSON.java

```diff
@@ -0,0 +1,310 @@
+            }
+        }
+
+        // ForwardAssignment
+        for (Node node : nodesMap.values()) {
+            var behaviour = node.getBehaviour();
+            for (Pin pin : behaviour.getOutPin()) {
+                var assignment = ddFactory.createForwardingAssignment();
+                assignment.setOutputPin(pin);
+                assignment.getInputPins().addAll(behaviour.getInPin());
+                behaviour.getAssignment().add(assignment);
+            }
+        }
+
+        return new CompleteDFD(dfd, dd);
+    }
+
+    public List<Label> createLabels(List<String> labelNames, DataDictionary dd, LabelType annotation) {
+        List<Label> labels = new ArrayList<>();
+        for (String labelName : labelNames) {
+            if (labelMap.containsKey(labelName)) {
+                labels.add(labelMap.get(labelName));
+            } else {
+                Label label = ddFactory.createLabel();
+                label.setEntityName(labelName);
+                annotation.getLabel().add(label);
+                labels.add(label);
+                labelMap.put(labelName, label);
+            }
+        }
+        return labels;
+    }
+
+    public CompleteDFD processWeb(DFD webdfd) {
+        nodesMap = new HashMap<String, Node>();
+        Map<String, Node> pinToNodeMap = new HashMap<String, Node>();
+        Map<String, Pin> pinMap = new HashMap<String, Pin>();
+        Map<String, Label> idToLabelMap = new HashMap<>();
+        Map<Node, Map<Pin, String>> nodeOutpinBehavior = new HashMap<>();
+
+        DataFlowDiagram dfd = dfdFactory.createDataFlowDiagram();
+        DataDictionary dd = ddFactory.createDataDictionary();
+
+        for (WebLabelType webLabelType : webdfd.labelTypes()) {
+            LabelType labelType = ddFactory.createLabelType();
+            labelType.setEntityName(webLabelType.name());
+            labelType.setId(webLabelType.id());
+            for (Value value : webLabelType.values()) {
+                Label label = ddFactory.createLabel();
+                label.setEntityName(value.text());
+                label.setId(value.id());
+                labelType.getLabel().add(label);
+                idToLabelMap.put(label.getId(), label);
+            }
+            dd.getLabelTypes().add(labelType);
+
+        }
+
+        for (Child child : webdfd.model().children()) {
+            String[] type = child.type().split(":");
+            String name = child.text();
+
+            if (type[0].equals("node")) {
+                Node node;
+                if (type[1].equals("function")) {
+                    node = dfdFactory.createProcess();
+                } else if (type[1].equals("storage")) {
+                    node = dfdFactory.createStore();
+                } else if (type[1].equals("input-output")) {
+                    node = dfdFactory.createExternal();
+                } else {
+                    node = null;
+                    System.out.println("Error");
+                }
```

Comment:
Can be replaced by a switch case. Also, Error Message on System.out should be logged to logger.error instead

Reply:
Also: In the case of an error, the method will continue and thus crash in the next line, because node is null